### PR TITLE
fix(index): bind Windows source checks to descriptor metadata

### PR DESCRIPTION
**Pull request by aimasteracc**
## Summary
- The fresh-clone secrets-baseline contract now transfers real Git history without checking out unrelated source files. It explicitly checks out only .secrets.baseline, preserves reviewed ancestry and removes the remote before validation.
- Add a narrowly path-routed/manual Windows 3.11/3.13 diagnostic: 20 iterations of the original three node IDs, default xdist and timeout settings, no pytest reruns, pinned head SHA and locked environment.
- Preserve every log, JUnit result and exit code; any failed, missing or skipped case fails qualification. Later passing attempts cannot erase the first failure.

## Evidence
- New checkout-scope assertion failed against the full-checkout implementation, then passed after the minimal change.
- Local focused: 87 passed, 1 existing skip; TSA focused contract: 85 passed, 1 existing skip.
- Quick gate: 2006 passed, 28 skipped.
- actionlint and git diff --check passed.

## Limits
This removes unnecessary checkout work; it does not yet prove the root cause of every Windows worker termination. Native repeated diagnostics must run before drawing conclusions. This PR does not close #1373 or #1356 merely because a later run passes. No additional skip, raised timeout or reduced worker count.

Related: #1373, #1356.

### CODE REVIEW

**Comment by aimasteracc** [comment]
### Root-cause evidence and fix (8b35f6c1)

Native Windows 3.13 artifacts from run 34079709721 contain the full metadata records. In failing iterations, lstat and fstat agree on device/inode/mode/size/mtime/attributes but differ on ctime; both descriptor fstat observations agree with each other. The old cross-interface comparison therefore rejected stable bytes.

The fix compares the complete read-before/read-after metadata on the same descriptor. Windows also retains the pre-open size/mtime/attribute checks and existing device/inode/mode binding; POSIX retains the original full path-to-open comparison. No sleep, retry, or later-path-stat recovery was added.

Nine boundary cases include the converse safety failure: a descriptor ctime change must remain unsafe even when its final value equals the older path observation. Local 182 associated tests and patch coverage passed; quick gate 2016 passed, 28 existing skips. The new candidate is now running the same native 20-iteration diagnostic. Original failing artifacts remain preserved; no issue is being closed based on an isolated green attempt.

File: .github/workflows/windows-trust-diagnostics.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+              foreach ($suite in @($junit.testsuites.testsuite)) {
+                $executed += [int]$suite.tests
+                $skipped += [int]$suite.skipped
+                if (([int]$suite.failures + [int]$suite.errors) -ne 0) { $code = 1 }
+              }
+            } catch {
+              $xmlError = $_.Exception.Message
+              $code = 1
+            }
+            if ($executed -ne 3 -or $skipped -ne 0) { $code = 1 }
```
[the remark this answers is not in the document]
Comment (aimasteracc) [reply .github/workflows/windows-trust-diagnostics.yml:80]: Fixed: raw pytest LASTEXITCODE is now preserved as exit_code. JUnit/count/skip qualification changes only qualification_exit_code and the job aggregate; neither missing artifacts nor exit 2/3/5 are rewritten in the raw evidence.
